src/lib/utilities.ts

```diff
@@ -0,0 +1,39 @@
+import supportedChains from "./chains";
+import { IChainData } from "./types";
+
+export function getChainData(chainId?: number): IChainData | null {
+  if (!chainId) {
+    return null;
+  }
+  const chainData = supportedChains.filter(
+    (chain: any) => chain.chain_id === chainId
+  )[0];
+
+  if (!chainData) {
+    throw new Error("ChainId missing or not supported");
+  }
+
+  const API_KEY = "460f40a260564ac4a4f4b3fffb032dad";
```

Comment:
could this be added to the `constants.ts` too? or is the plan to keep both INFURA_ID AND API_KEY secret later in a .env file?